staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/webhook_converter.go

```diff
@@ -0,0 +1,179 @@
+	unstructIn, ok := in.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("input type %T in not valid for unstructured conversion", in)
+	}
+
+	unstructOut, ok := out.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("output type %T in not valid for unstructured conversion", out)
+	}
+
+	outGVK := unstructOut.GroupVersionKind()
+	if !c.validVersions[outGVK.GroupVersion()] {
+		return fmt.Errorf("request to convert CRD from an invalid group/version: %s", outGVK.String())
+	}
+	inGVK := unstructIn.GroupVersionKind()
+	if !c.validVersions[inGVK.GroupVersion()] {
+		return fmt.Errorf("request to convert CRD to an invalid group/version: %s", inGVK.String())
+	}
+
+	unstructOut.SetUnstructuredContent(unstructIn.UnstructuredContent())
+	_, err := c.ConvertToVersion(unstructOut, outGVK.GroupVersion())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createConversionReview(obj runtime.Object, apiVersion string) *v1beta1.ConversionReview {
+	_, isList := obj.(*unstructured.UnstructuredList)
+	return &v1beta1.ConversionReview{
+		Request: &v1beta1.ConversionRequest{
+			Object:     runtime.RawExtension{Object: obj},
+			APIVersion: apiVersion,
+			UID:        uuid.NewUUID(),
+			IsList:     isList,
+		},
+		Response: &v1beta1.ConversionResponse{},
+	}
+}
+
+func (c *webhookConverter) ConvertToVersion(in runtime.Object, target runtime.GroupVersioner) (runtime.Object, error) {
+	fromGVK := in.GetObjectKind().GroupVersionKind()
+	toGVK, ok := target.KindForGroupVersionKinds([]schema.GroupVersionKind{fromGVK})
+	if !ok {
+		// TODO: should this be a typed error?
+		return nil, fmt.Errorf("%v is unstructured and is not suitable for converting to %q", fromGVK, target)
+	}
+	if !c.validVersions[toGVK.GroupVersion()] {
+		return nil, fmt.Errorf("request to convert CRD to an invalid group/version: %s", toGVK.String())
+	}
+	if !c.validVersions[fromGVK.GroupVersion()] {
+		return nil, fmt.Errorf("request to convert CRD from an invalid group/version: %s", fromGVK.String())
+	}
+	if fromGVK == toGVK {
+		// No conversion is required
+		return in, nil
+	}
+
+	request := createConversionReview(in, toGVK.GroupVersion().String())
+	response := &v1beta1.ConversionReview{}
+	ctx := context.TODO()
+	if err := c.restClient.Post().Context(ctx).Body(request).Do().Into(response); err != nil {
+		// TODO: Return a webhook specific error to be able to convert it to meta.Status
+		return nil, fmt.Errorf("calling to conversion webhook failed for %s, with error=%s", c.name, err)
+	}
+
+	if response.Response == nil {
+		// TODO: Return a webhook specific error to be able to convert it to meta.Status
+		return nil, fmt.Errorf("conversion webhook response was absent for %s", c.name)
+	}
+
+	if response.Response.Result != nil {
+		// TODO return status message as error
+		return nil, fmt.Errorf("conversion request failed for %v: %v", in.GetObjectKind(), response.Response.Result.Message)
+	}
+
+	ret := response.Response.ConvertedObject.Object
```

Comment:
do we have to check that `ConvertedObject` is not nil?

Reply:
ConvertedObject is not a pointer and required. The object inside can be nil though.